### PR TITLE
fix(resolve): report unresolved imports firstly

### LINKED ISSUE #81413: Rustc shows unnecessary big error when glob reexport failed
I tried this code:

```rust
pub const ITEM: Item = Item;

pub struct Item;

pub fn item() {}

// Glob (*) is required to cause the error
pub use doesnt_exist::*;

mod a {
    use crate::{item, Item, ITEM};
}

mod b {
    use crate::item;
    use crate::Item;
    use crate::ITEM;
}

mod c {
    use crate::item;
}
```
([play](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=d7d38220140a6793756d30e20fdbdc22))

I expected to see an error that only says that "`doesnt_exist` is unresolved".

But the error says that all uses of items from the module with glob import (`crate` in this case) are unresolved too:

```text
error[E0432]: unresolved imports `doesnt_exist`, `crate::item`, `crate::Item`, `crate::ITEM`, `crate::item`, `crate::Item`, `crate::ITEM`, `crate::item`
  --> src/lib.rs:41:9
   |
41 | pub use doesnt_exist::*;
   |         ^^^^^^^^^^^^ use of undeclared crate or module `doesnt_exist`
...
44 |     use crate::{item, Item, ITEM};
   |                 ^^^^  ^^^^  ^^^^
...
48 |     use crate::item;
   |         ^^^^^^^^^^^
49 |     use crate::Item;
   |         ^^^^^^^^^^^
50 |     use crate::ITEM;
   |         ^^^^^^^^^^^
...
54 |     use crate::item;
   |         ^^^^^^^^^^^
```

This may lead to extremely long and unreadable errors. The issue was originally found while developing `teloxide` where it caused [1500+ lines error](https://gist.github.com/WaffleLapkin/0283b070e38fed0affdf5b787b0010d3).

### Meta

`rustc --version --verbose`:
```
rustc 1.51.0-nightly (c97f11af7 2021-01-10)
binary: rustc
commit-hash: c97f11af7bc4a6d3578f6a953be04ab2449a5728
commit-date: 2021-01-10
host: x86_64-unknown-linux-gnu
release: 1.51.0-nightly
```

### PR DESCRIPTION
**Pull request by bvanjoi**
Fixes #81413

An easy fix, r? @petrochenkov

### CODE REVIEW

File: compiler/rustc_resolve/src/imports.rs [hunk 1]
```diff
@@ -463,6 +462,10 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
         }
     }
 
+    fn has_resolution(&self, bindings: &PerNS<Cell<Option<NameBinding<'a>>>>) -> bool {
```
Comment (petrochenkov) [inline compiler/rustc_resolve/src/imports.rs:465]:
```suggestion
    fn has_resolution(bindings: &PerNS<Cell<Option<NameBinding<'a>>>>) -> bool {
```
`self` is unused.

File: compiler/rustc_resolve/src/imports.rs [hunk 2]
```diff
@@ -509,6 +512,10 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                     prev_root_id = import.root_id;
                 }
             } else if is_indeterminate {
+                if let ImportKind::Single { target_bindings, .. } = &import.kind
+                && self.has_resolution(&target_bindings) {
+                    continue;
```
Comment (petrochenkov) [inline compiler/rustc_resolve/src/imports.rs:517]: I don't understand why it's legal to skip the error here.
What guarantees that the compilation won't be successful?

I also don't understand why `has_resolution` was selected as a criterion for skipping errors.
From what I understand, `import_dummy_binding` above should guarantee that all indeterminate bindings will have resolutions.
So this line just skips errors from *all* indeterminate single imports?

Comment (bvanjoi) [reply compiler/rustc_resolve/src/imports.rs:517]: Yep, I had concerned it too, I think I should deal to the error output rather than skip errors.

Comment (bvanjoi) [reply compiler/rustc_resolve/src/imports.rs:517]: > So this line just skips errors from all indeterminate single imports?

No, `import_dummy_binding` only define dummy binding for resolution but not modify the value of `target_bindings`.

**Comment by bvanjoi** [comment]
I have adopted a more intuitive approach: initially, an ﻿`unresolved_import` is reported, followed by ﻿`indeterminate_imports` if no errors are present. 

@rustbot ready

**Comment by petrochenkov** [comment]
This is certainly better.
@bors r+

**Pull request merged**